goal_src/jak2/engine/target/gun/gun-h.gc

```diff
@@ -5,5 +5,322 @@
+    (the-as
+      symbol
+      (and (logtest? (-> arg0 game features) 32)
+           (>= (- (-> pp clock frame-counter) (-> arg0 gun gun-time)) (seconds 0.1))
+           (zero? (logand (focus-status fs1 fs6 fs18 fs21 fs22 fs30) (-> arg0 focus-status)))
+           (zero? (logand (surface-flag gun-inactive gun-hide gun-off) (-> arg0 control unknown-surface00 flags)))
+           (zero? (logand (state-flags sf18) (-> arg0 state-flags)))
+           (logtest? (logand (-> *setting-control* user-current features) 960) (-> arg0 game features))
+           (or (zero? (logand (-> arg0 control unknown-surface00 flags) (surface-flag duck))) (can-exit-duck?))
+           (or (zero? (logand (focus-status fs20) (-> arg0 focus-status))) (-> arg0 pilot gun?))
+           (or arg1
+               (nonzero? (-> arg0 gun using-gun-type))
+               (begin
+                 (set! v1-36 (and (cpad-hold? (-> arg0 control unknown-cpad-info00 number) r1)
+                                  (zero? (logand (-> arg0 focus-status) (focus-status fs7)))
+                                  (begin (set! v1-36 #t) (set! (-> arg0 gun latch?) v1-36) v1-36)
+                                  )
+                       )
+                 v1-36
+                 )
+               (-> arg0 gun latch?)
+               )
+           (not (-> arg0 skel top-anim frame-group))
+           (>= (- (-> pp clock frame-counter) (the-as int (-> arg0 control unknown-dword01))) (seconds 0.1))
+           )
+      )
+    )
+  )
+
+(defun using-gun? ((arg0 target))
+  "Are we using a gun?"
+  (nonzero? (-> arg0 gun gun-type))
+  )
+
+(defun enabled-gun? ((arg0 target))
+  (nonzero? (-> arg0 gun using-gun-type))
+  )
+
+(defun gun->ammo ((arg0 pickup-type))
+  "Get ammo type based on gun."
+  (case arg0
+    (((pickup-type eco-yellow))
+     13
+     )
+    (((pickup-type eco-red))
+     14
+     )
+    (((pickup-type eco-blue))
+     15
+     )
+    (((pickup-type eco-dark))
+     16
+     )
+    (else
+      0
+      )
+    )
+  )
+
+(defmethod get-gun-ammo fact-info-target ((obj fact-info-target) (arg0 target))
+  "Get ammo count for current gun."
+  (let ((current-gun (gun->ammo (-> (the-as target (-> obj process)) gun gun-type))))
+    (if (zero? current-gun)
+        0.0
+        (-> (the-as target (-> obj process)) game gun-ammo (+ current-gun -13))
+        )
+    )
+  )
+
+;; failed to figure out what this is:
+(let ((a0-5 (new 'static 'skeleton-group
+              :name "skel-gun"
+              :extra #f
+              :info #f
+              :art-group-name "gun"
+              :bounds (new 'static 'vector :w 6144.0)
+              :version #x7
+              :shadow #xb
+              :sort 1
+              :origin-joint-index 3
+              )
+            )
+      )
+  (set! (-> a0-5 jgeo) 9)
+  (set! (-> a0-5 janim) -1)
+  (set! (-> a0-5 mgeo 0) 10)
+  (set! (-> a0-5 data 12) (the-as art-element 4095996000.0))
+  ((method-of-object a0-5 skeleton-group-method-15))
```

Comment:
This weird looking thing is a method call if the method hasn't been set up at all yet.
I looked ahead at this method, and it seems to add this skeleton group to the currently loading level.  So you could change `deftype skeleton-group`'s `:methods` to:
```
  (:methods
    (add-to-loading-level (_type_) none 15)
    )
```